clippy_lints/src/misc.rs

```diff
@@ -493,6 +499,25 @@ fn is_allowed<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, expr: &'tcx Expr) -> bool {
     }
 }
 
+// Return true if `expr` is the result of `signum()` invoked on a float value.
+fn is_signum(cx: &LateContext<'_, '_>, expr: &Expr) -> bool {
+    // The negation of a signum is still a signum
```

Comment:
Nice, good catch!

Lint idea (for another PR):
```rust
// for
f1.signum() == -f2.signum()
// suggest
f1.signum() != f2.signum()
```

Same goes for `-sig==+sig` and `-sig==-sig` and the `!=` variants.

Reply:
Thanks! I'm not sure that's the best way to simplify `f1.signum() == -f2.signum()`, since if both `f1` and `f2` are 0, then also `f1.signum() == f2.signum() == 0`.

Reply:
Oh your right, in this case we can't really lint this. Never mind then :D